### PR TITLE
Vision image properties

### PR DESCRIPTION
**Pull request by daspecster**
Based from #2319

Adds the image properties detection API for Vision.

This adds a fairly large `ImagePropertiesAnnotation` with several classes to absorb the color data in the response.

✅ Also after #2319 is merged I will clean up the weird commit history.

### CODE REVIEW

**Comment by daspecster** [comment]
@tseaver, I cleaned this up so it should be ready to checkout as well.

File: google/cloud/vision/color.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        red = response['red']
+        green = response['green']
+        blue = response['blue']
+        alpha = response.get('alpha')
+
+        return cls(red, green, blue, alpha)
+
+    @property
+    def red(self):
+        """Red color.
```
Comment (tseaver) [inline google/cloud/vision/color.py]: Maybe "Red component of the color"?

File: google/cloud/vision/color.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+        :rtype: int
+        :returns: Red RGB value.
+        """
+        return self._red
+
+    @property
+    def green(self):
```
Comment (tseaver) [inline google/cloud/vision/color.py]: Likewise

File: google/cloud/vision/color.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        """Green color.
+
+        :rtype: int
+        :returns: Green RGB value.
+        """
+        return self._green
+
+    @property
```
Comment (tseaver) [inline google/cloud/vision/color.py]: Likewise

File: google/cloud/vision/color.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        """
+        return self._colors
+
+
+class Color(object):
+    """Representation of RGBA color information.
+
+    :type red: int
+    :param red: The amount of red in the color as a value in the interval
+                [0, 1].
```
Comment (tseaver) [inline google/cloud/vision/color.py]: That doesn't match the type:  (unless we are talking about 6-bit color :)

File: google/cloud/vision/color.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+
+class ColorInformation(object):
+    """Representation of color information from API response.
+
+    :type color: :class:`~google.cloud.vision.color.Color`
+    :param color: RGB components of the color.
+
+    :type score: float
+    :param score: Image-specific score for this color. Value in range [0, 1].
```
Comment (tseaver) [inline google/cloud/vision/color.py]: What does it mean?

File: unit_tests/vision/_fixtures.py [hunk 6]
```diff
@@ -1569,6 +1673,18 @@
...
+        {
+            'safeSearchAnnotation': {
+                'adult': 'VERY_UNLIKELY',
+                'spoof': 'UNLIKELY',
+                'medical': 'POSSIBLE',
+                'violence': 'VERY_UNLIKELY'
+            }
+        }
+    ]
+}
```
Comment (tseaver) [inline unit_tests/vision/_fixtures.py]: Is this a "stray" diff from the earlier PR for "safe" annotations?

File: google/cloud/vision/color.py (hunk 4 above)
Comment (daspecster) [inline google/cloud/vision/color.py]: Ha! I copied it from the API docs [here](https://cloud.google.com/vision/reference/rest/v1/images/annotate#color).

Can't believe I missed that.

File: unit_tests/vision/_fixtures.py (hunk 6 above)
Comment (daspecster) [inline unit_tests/vision/_fixtures.py]: Ah yes. Stacking these PR's didn't workout as well as I had hoped.

File: google/cloud/vision/color.py (hunk 5 above)
Comment (daspecster) [inline google/cloud/vision/color.py]: So I _think_ it represents how dominant the color is in the image. I can't seem to find documentation to back that up though.

File: unit_tests/vision/_fixtures.py (hunk 6 above)
Comment (daspecster) [inline unit_tests/vision/_fixtures.py]: Yeah, basing off the other branches didn't work exactly how I expected it.

**Comment by daspecster** [comment]
@tseaver, updated everything except the `score` docstring. That value is copied from their docs, but I can't seem to find a better explanation of it.

**Comment by tseaver** [comment]
LGTM

**Pull request merged**